rotkehlchen/exchanges/coinbase.py

```diff
@@ -160,21 +161,117 @@ def __init__(
             secret=secret,
             database=database,
         )
+        self.is_legacy_api_key=self.determine_key_type(api_key)
         self.apiversion = 'v2'
         self.base_uri = 'https://api.coinbase.com'
         self.msg_aggregator = msg_aggregator
-        self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
-        self.session.headers.update({'CB-VERSION': CB_VERSION})
+        self.host = 'api.coinbase.com'
+
+        
+        if self.is_legacy_api_key:
+            self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
+            self.session.headers.update({'CB-VERSION': CB_VERSION})
+
+    def determine_key_type(self, api_key)-> bool:
+        """
+        Determines if the provided API key and secret are legacy or new.
+        Returns True if legacy, False if new.
+        """
```

Comment:
Use self-explanatory function names

```suggestion
    def is_legacy_key(self, api_key)-> bool:
        """Determines if the provided API key is a legacy one"""
```